### ISSUE TITLE
Warn of unqoted variables in case.

### ISSUE THREAD

**Issue by SoptikHa2** [koalaman/shellcheck#1682]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/zsh # Bash works the same way
$choice="[[]]"

$answer_1="abc"
$answer_2="[[]]"
$answer_3="def"

if [ "$choice" = "$answer_2" ]; then
  echo "They are equal."
fi

case $choice in
  $answer_1)
    echo "1"
  ;;
  $answer_2)
    echo "2"
  ;;
  $answer_3)
    echo "3"
esac
```

Output of this script is:
```
They are equal.
```

While I would expect output:
```
They are equal.
2
```

The solution is to quote variables in `case`, like this:
```sh
case "$choice" in
    "$answer_1")
   #...
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Shellcheck should warn user in the same way as when we use unquoted variables, which should be warning [SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086).

**Comment by koalaman** [comment]
You're right, this is the same effect as seen in [SC2053](https://github.com/koalaman/shellcheck/wiki/SC2053), and there's now a [new warning](https://github.com/koalaman/shellcheck/wiki/SC2254) for this. Thanks!
